### DISCUSSION TITLE
How can I ignore some files in the pages directory to prevent them from becoming routes?

### DISCUSSION THREAD

**Discussion by s3xysteak** [nuxt/nuxt#28084] [Questions]
If I have a component, which was only used in the page, I want to place it just besides the file which use it.
- pages
  - index.vue
  - other-route.vue
  - Foo.vue
  - Bar.vue
  
If I just put these components in `/Component`, It feels so annoyed that so many component names in intellisense but most of them are only used in one or two page.

**Comment by xjccc** [comment]
Look at [ignore](https://nuxt.com/docs/api/nuxt-config#ignore) have a try?

- pages
 - test
  -  test/index.vue
  - test/test.comp.vue
 
```
  ignore: ['**/*.comp.vue']
```

```
// test/index.vue
<template>
  <div class="container">
    测试一下
    <Test />
  </div>
</template>
<script setup lang="ts">
  import Test from './test.comp.vue'
</script>
<style lang="less" scoped></style>

```

```
// test/test.comp.vue

<template>
  <div class="container">我是测试的组件</div>
</template>
<script setup lang="ts"></script>
<style lang="less" scoped></style>

```

**Reply by s3xysteak** [reply]
It not works 😢  .
With `ignore: ['./pages/**/*.comp.vue']`, I can still access it by the route `/test.comp`.

**Reply by xjccc** [reply]
https://stackblitz.com/edit/nuxt-starter-z1a8vr?file=pages%2Findex.vue,pages%2Ftest%2Findex.vue

I cant access the route.

**Reply by s3xysteak** [reply]
> https://stackblitz.com/edit/nuxt-starter-z1a8vr?file=pages%2Findex.vue,pages%2Ftest%2Findex.vue
> 
> I cant access the route.

Thank you! I found out that I am using `'./pages/**/*.comp.vue'` instead of `'**/*.comp.vue'` in nuxt.config, which caused the difference. Why doesn't `'./pages/**/*.comp.vue'` match?
